components/server/src/user/newsletter-subscription-controller.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import * as express from 'express';
+import { inject, injectable } from "inversify";
+import { UserDB } from "@gitpod/gitpod-db/lib";
+import { IAnalyticsWriter } from "@gitpod/gitpod-protocol/lib/analytics";
+
+@injectable()
+export class NewsletterSubscriptionController {
+    @inject(UserDB) protected readonly userDb: UserDB;
+    @inject(IAnalyticsWriter) protected readonly analytics: IAnalyticsWriter;
+
+    get apiRouter(): express.Router {
+        const router = express.Router();
+
+        router.get("/unsubscribe", async (req: express.Request, res: express.Response) => {
```

Comment:
we would like to use this API to subscribe newsletter users so that they are automatically attached to their user accounts if a the email for the newsletter signup matches any user's primary email. is it ok for you to extend the scope of this PR to include a `/subscribe` endpoint that signs up a known or unknown user for a specified subscription list?

Reply:
I would prefer to keep the existing scope of this ticket for reasons of: 1) planning and 2) transparency :) When tickets are approved and put to in progress, I assume there are already plans and estimations of how much effort the current ticket/board would take. Changing the scope might change the planning for the rest of the board. Secondly, a changing scope almost at the end of the work might not be known to the stakeholders of the board.
However, I am still learning how the team usually manages their projects so I will tag @JanKoehnlein here for his thoughts.

Reply:
Let's open another ticket for @jakobhero's request. I feel like it has to be discussed in terms of security and GDPR as well.